ethereum/blockwatch/block_watcher.go

```diff
@@ -37,50 +37,54 @@ const (
 // Event describes a block event emitted by a Watcher
 type Event struct {
 	Type        EventType
-	BlockHeader *meshdb.MiniHeader
+	BlockHeader *miniheader.MiniHeader
+}
+
+// Stack defines the interface a stack must implement in order to be used by
+// OrderWatcher for block header storage
+type Stack interface {
+	Pop() (*miniheader.MiniHeader, error)
+	Push(*miniheader.MiniHeader) error
+	Peek() (*miniheader.MiniHeader, error)
+	Inspect() ([]*miniheader.MiniHeader, error)
```

Comment:
IIRC `Inspect` is only used in tests and debugging. I wonder if we should leave the `Inspect` method out of the interface and just upcast if we need to use it in tests?

```go
// In the tests we always know that we're using SimpleStack so we can call the Inspect method.
w.stack.(*SimpleStack).Inspect()
```

This would reduce the amount of code needed for external developers to write a `Stack` implementation, and in particular would make it easier/more efficient to implement `Stack` using a linked list.

Reply:
It's actually also used in `BlockWatcher` on line 330. What I've done instead is renamed it to `PeekAll`. The Stack implementation doesn't need to be that performant since it only stored up to 20 blocks. It won't ever make sense to store much more then that.